src/lib/RpcApi.ts

```diff
@@ -167,6 +184,10 @@ export default class RpcApi {
         };
     }
 
+    private _exportStateToHistory() {
+        history.replaceState({ ...history.state, exportedState: this._exportState() }, '');
+    }
```

Comment:
Would it also be possible to not store in the history object, but in localstorage and replace it there on every navigation? Would that be able to fulfill the same purpose?

Reply:
I don't think that the use of either `localStorage` (or `sessionStorage`) would be recommendable.
`localStorage` would be persistent which we don't need and would thus be more wasteful. `localStorage` is also shared across multiple tabs which potentially leads to problems.
When using `sessionStorage` we only have one store instance we'd be overwriting all the time, thus on back navigation we can't restore the state we had at that specific page / history entry. This becomes for example problematic when executing multiple redirect requests in a row in the same tab. If the user navigates back to the previous request, we wouldn't be able to resume the state for that request (if it doesn't come from other sources like the request).

We might be able to use the session storage if accessing the state in the session storage by request id. But we wouldn't really win anything by doing so and the use of the history state is more general and cleaner.